lib/Retcon/Network/Server.hs

```diff
@@ -102,13 +106,17 @@ spawnServer cfg n = do
         sock   <- socket ctx Rep
         bind sock zmq_conn
         db     <- initBackend (PGOpts pg_conn)
-        return $  ServerState ctx sock cfg db
+        -- Setup ekg
+        ekgStore  <- initialiseMeters cfg
+        ekgServer <- Ekg.forkServerWith ekgStore "localhost" 8888
+        return $  ServerState ctx sock cfg db ekgServer
 
     stop :: ServerState -> IO ()
     stop state = do
         closeBackend $ state ^. serverStore
         close        $ state ^. serverSocket
         term         $ state ^. serverContext
+        killThread   $ Ekg.serverThreadId $ state ^. serverEkgServer
```

Comment:
Apparently this is the kosher way? pretty sure the exception will only be re-raised if we link the threads?

From ekg docs:
serverThreadId :: Server -> ThreadId
The thread ID of the server. You can kill the server by killing this thread (i.e. by throwing it an asynchronous exception.)